x/stakeibc/keeper/msg_server_change_validator_weight.go

```diff
@@ -0,0 +1,30 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Stride-Labs/stride/x/stakeibc/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func (k msgServer) ChangeValidatorWeight(goCtx context.Context, msg *types.MsgChangeValidatorWeight) (*types.MsgChangeValidatorWeightResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	hostZone, found := k.GetHostZone(ctx, msg.HostZone)
+	if !found {
+		k.Logger(ctx).Error(fmt.Sprintf("Host Zone %s not found", msg.HostZone))
+		return nil, types.ErrInvalidHostZone
+	}
+	validators := hostZone.Validators
+	for _, validator := range validators {
```

Comment:
Can / should we store validators in a map and index them with address, rather than iterate them every time? Might not a be a big difference, maybe best to add a TODO for later